tests/btf.cpp

```diff
@@ -0,0 +1,529 @@
+#include <fstream>
+#include <string>
+#include <unordered_set>
+
+#include "btf/btf.h"
+#include "btf/helpers.h"
+#include "gtest/gtest.h"
+
+namespace bpftrace::test::btf {
+
+using namespace bpftrace::btf;
+
+static std::string to_str(const AnyType &type)
+{
+  std::stringstream out;
+  out << type;
+  return out.str();
+}
+
+TEST(btf, to_str)
+{
+  Types btf;
+  auto int8 = btf.add<Integer>("int8", 1, 1);
+  ASSERT_TRUE(bool(int8));
+  EXPECT_EQ(to_str(AnyType(*int8)), "int8");
+}
+
+TEST(btf, kernel_types)
+{
+  std::ifstream file("/sys/kernel/btf/vmlinux");
+  std::stringstream buffer;
+  buffer << file.rdbuf();
+  std::string content = buffer.str();
+  auto btf = Types::parse(content.c_str(), content.size());
+  ASSERT_TRUE(bool(btf));
+
+  // Want to see all the functions available?
+  std::unordered_set<std::string> funcs;
+  for (const Function &func : *btf | filter<Function>()) {
+    EXPECT_FALSE(funcs.contains(func.name()));
+    funcs.insert(func.name());
+  }
+
+  // Let's just look at the variables.
+  std::unordered_set<std::string> vars;
+  for (const Var &var : *btf | filter<Var>()) {
+    EXPECT_FALSE(vars.contains(var.name()));
+    vars.insert(var.name());
+  }
+}
+
+TEST(btf, map_types)
+{
+  Types btf;
+  auto int8 = btf.add<Integer>("int8", 1, 1);
+  ASSERT_TRUE(bool(int8));
+  EXPECT_EQ(to_str(AnyType(*int8)), "int8");
+
+  auto m = createMap(btf,
+                     MapInfo{
+                         .map_type = BPF_MAP_TYPE_HASH,
+                         .key = *int8,
+                         .value = *int8,
+                         .nr_elements = 1,
+                     });
+  ASSERT_TRUE(bool(m));
+  EXPECT_EQ(to_str(AnyType(*m)), "struct (anon)");
```

Comment:
We can't detect it's a map and print something nicer?

Reply:
There's another test for `mapInfo` which can extract this from the type. The default `operator<<` will print the C literal type, but you're right -- I can adjust this and do something nicer.